test/e2e/operator_upgrade_test.go

```diff
@@ -0,0 +1,113 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	. "github.com/onsi/gomega"
+	"strings"
+	"testing"
+
+	hyperShiftAPI "github.com/openshift/hypershift/api/v1beta1"
+	"github.com/openshift/hypershift/cmd/install"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	awsCredentialsSecretKey           = "credentials"
+	awsRegion                         = "us-east-1"
+	hyperShiftCredentialsFilePath     = "/etc/hypershift-pool-aws-credentials/credentials"
+	hyperShiftExternalDNSDomainFilter = "service.ci.hypershift.devcluster.openshift.com"
+	hyperShiftOIDCBucket              = "hypershift-ci-oidc"
+	latestHyperShiftOperatorImage     = "quay.io/hypershift/hypershift-operator:latest"
+)
+
+// TestOperatorUpgrade implements a test that creates a guest cluster, upgrades the HyperShift operator, and
+// ensures there were no restarted/crashed pods in the guest cluster during the upgrade.
+func TestOperatorUpgrade(t *testing.T) {
+	t.Parallel()
+	g := NewWithT(t)
+
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	client, err := e2eutil.GetClient()
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get k8s client")
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+	zones := strings.Split(globalOpts.configurableClusterOptions.Zone.String(), ",")
+	if len(zones) >= 3 {
+		// CreateCluster also tests multi-zone workers work properly if a sufficient number of zones are configured
+		t.Logf("Sufficient zones available for InfrastructureAvailabilityPolicy HighlyAvailable")
+		clusterOpts.AWSPlatform.Zones = zones
+		clusterOpts.InfrastructureAvailabilityPolicy = string(hyperShiftAPI.HighlyAvailable)
+		clusterOpts.NodePoolReplicas = 1
+	}
+
+	// Install the latest version of the HyperShift operator
+	err = installHyperShiftOperatorVersion(ctx, latestHyperShiftOperatorImage)
```

Comment:
All the test suite runs against the same HO and seems that you are overriding that so this would affect to all the test suite in a non deterministic way which is not what we want.